<!WRONG_MODIFIER_TARGET!>enum<!> interface Some {
    // Enum part
    <!ABSTRACT_MEMBER_NOT_IMPLEMENTED, ENUM_ENTRY_USES_DEPRECATED_OR_NO_DELIMITER!>D<!>

    // Interface like part
    fun test()
    val foo: Int
}